jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/ddl/TestDBPlatformHelper.java

```diff
@@ -51,19 +51,47 @@ public void test() {
     @Test
     public void testDB2ZOS() {
         //Returned from jcc driver
-        Assert.assertEquals(DB2ZPlatform.class.getName(), getPlatformClass("DB2", "DSN10015"));
+        Assert.assertEquals(org.eclipse.persistence.platform.database.DB2ZPlatform.class.getName(), getPlatformClass("DB2", "10", "DSN10015"));
     }
 
     @Test
     public void testDB2I() {
         //Returned from jcc driver (DRDA)
-        Assert.assertEquals(DB2MainframePlatform.class.getName(), getPlatformClass("AS", "QSQ07020"));
+        Assert.assertEquals(org.eclipse.persistence.platform.database.DB2MainframePlatform.class.getName(), getPlatformClass("AS", "10", "QSQ07020"));
 
         //Returned from type 2 native driver & type 4 open source driver (non-DRDA)
-        Assert.assertEquals(DB2MainframePlatform.class.getName(), getPlatformClass("DB2 UDB for AS/400", "07.02.0000 V7R2m0"));
+        Assert.assertEquals(org.eclipse.persistence.platform.database.DB2MainframePlatform.class.getName(), getPlatformClass("DB2 UDB for AS/400", "07.02.0000", "V7R2m0"));
     }
 
-    private String getPlatformClass(String productName, String productVersion){
-        return DBPlatformHelper.getDBPlatform(productName + productVersion, log);
+    @Test
+    public void testMySQL() {
+        //Returned from jcc driver (DRDA)
+        Assert.assertEquals(org.eclipse.persistence.platform.database.MySQLPlatform.class.getName(), getPlatformClass("MySQL", "5", "5.5.5-10.1.33-MariaDB"));
+    }
+
+    @Test
+    public void testDerby() {
+        //Returned from jcc driver (DRDA)
+        Assert.assertEquals(org.eclipse.persistence.platform.database.JavaDBPlatform.class.getName(), getPlatformClass("Apache Derby", "10", "10.12.1.1 - (1704137)"));
+    }
+
+    @Test
+    public void testOracle() {
+        //Returned from jcc driver (DRDA)
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle9Platform", getPlatformClass("Oracle", "9", "Oracle Database 9c"));
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle10Platform", getPlatformClass("Oracle", "10", "Oracle Database 10c"));
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle11Platform", getPlatformClass("Oracle", "11", "Oracle Database 11c"));
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle12Platform", getPlatformClass("Oracle", "12", "Oracle Database 12c"));
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle18Platform", getPlatformClass("Oracle", "18", "Oracle Database 18c"));
```

Comment:
'org.eclipse.persistence.platform.database.oracle.OraclePlatform' is valid too, maybe wanna add it?

Reply:
Sure! What are the conditions to have OraclePlatform be set as the database platform? The only test case that comes to mind for me is when the version  < 9?

Reply:
the case used to be the fallback when either the exact version is not recognized or if 'oracle' jar is missing

Reply:
I'm not sure how to test for the latter, so I just added a test for the former. Does this work for you?

Reply:
Yes, sth like this is good enough.

BTW: NOT doing force-pushes makes it easier for others to find the exact diff between code which was already reviewed and new revision. Imagine the diff has 1k lines, one line needs to be changed and one has to verify that only that one particular line was changed...; one can use 'squash and merge' in github UI to make one commit out of the whole PR...

Reply:
I suppose that would make review easier, thank you for the suggestion.